### PR TITLE
GT-2044 persist apple user data

### PR DESCRIPTION
**Pull request by rachaelblue**

### CODE REVIEW

File: Sources/SocialAuthentication/AppleAuthentication/AppleAuthentication.swift [hunk 1]
```diff
@@ -71,10 +66,89 @@ extension AppleAuthentication {
...
+    }
+}
+
+// MARK: - User
+
+extension AppleAuthentication {
+    
+    public func getCurrentUserProfile() -> AppleUserProfile? {
+        
+        guard
```
Comment (levieggertcru) [inline Sources/SocialAuthentication/AppleAuthentication/AppleAuthentication.swift:85]: Hey @rachaelblue with using the guard here if any one of these values are null we won't get any profile data back. Any of these values could be null in ASAuthorizationAppleIDCredential. https://developer.apple.com/documentation/authenticationservices/asauthorizationappleidcredential

File: Sources/SocialAuthentication/AppleAuthentication/AppleAuthentication.swift [hunk 2]
```diff
@@ -71,10 +66,89 @@ extension AppleAuthentication {
...
+        }
+        
+        return AppleUserProfile(
+            email: email,
+            familyName: familyName,
+            givenName: givenName
+        )
+    }
+    
+    private func deletePersistedUser() {
```
Comment (levieggertcru) [inline Sources/SocialAuthentication/AppleAuthentication/AppleAuthentication.swift:100]: Hey @rachaelblue what are your thoughts on encapsulating this user persistence in another class?  Doing so would reduce some of the responsibilities and functionality in this class.  Thinking from a SOLID and unit tests point of view.

File: Sources/SocialAuthentication/AppleAuthentication/AppleAuthentication.swift (hunk 1 above)
Comment (rachaelblue) [reply Sources/SocialAuthentication/AppleAuthentication/AppleAuthentication.swift]: Changed the `AppleUserProfile` attributes to optionals.

File: Sources/SocialAuthentication/AppleAuthentication/AppleAuthentication.swift (hunk 2 above)
Comment (rachaelblue) [reply Sources/SocialAuthentication/AppleAuthentication/AppleAuthentication.swift]: Agreed, there's a lot of logic in there between keychain/user defaults.  Went ahead and moved it all into `AppleUserPersistentStore`.

**Review by levieggertcru — APPROVED** [review]

**Pull request merged**